.github/actions/build-container-bake/action.yml

```diff
@@ -0,0 +1,147 @@
+# - Maintenance overhead with duplicate build logic
+#
+# SOLUTION APPROACH:
+# - Single Docker Bake action building both targets simultaneously
+# - Registry-only caching strategy with unified cache reference
+# - zstd compression for optimal performance and storage efficiency
+# - Eliminates GitHub Actions cache to prevent contention
+#
+# CACHE STRATEGY:
+# - Uses only registry cache (type=registry) with 'main-cache' tag
+# - Leverages mode=max so production/runner layers contain development layers
+# - Single cache reference eliminates cache contention between targets
+# - zstd compression applied to all image outputs and cache configurations
+#
+# DESIGN DECISIONS FROM FEEDBACK:
+# - Removed GitHub Actions cache (type=gha) completely per review feedback
+# - Simplified to use only one cache image tag since production layer contains
+#   development layers when using mode=max
+# - Added compression=zstd to all image tags and cache-to configurations
+#   for better performance and storage efficiency
+#
+name: Build Container with Docker Bake
+description: Build and push Docker container images using Docker Bake with multiple targets and tags
+
+inputs:
+  registry:
+    description: 'Container registry URL'
+    required: true
+    default: 'ghcr.io'
+  image:
+    description: 'Base image name (without registry)'
+    required: true
+  platforms:
+    description: 'Comma-separated list of platforms to build for'
+    required: false
+    default: 'linux/amd64'
+  context:
+    description: 'Build context path'
+    required: false
+    default: './web'
+  dockerfile:
+    description: 'Path to Dockerfile'
+    required: false
+    default: './web/Dockerfile'
+  push:
+    description: 'Whether to push the built images'
+    required: false
+    default: 'true'
+
+outputs:
+  production-image:
+    description: 'Production image name with tag'
+    value: ${{ steps.set-outputs.outputs.production-image }}
+  development-image:
+    description: 'Development image name with tag'
+    value: ${{ steps.set-outputs.outputs.development-image }}
+  cache-image:
+    description: 'Cache image name with tag'
+    value: ${{ steps.set-outputs.outputs.cache-image }}
+  pr-cache-image:
+    description: 'PR cache image name with tag'
+    value: ${{ steps.set-outputs.outputs.pr-cache-image }}
+  digest:
+    description: 'Image digest'
+    value: ${{ steps.bake.outputs.digest }}
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Set lowercase image name
+      shell: bash
+      run: |
+        echo "IMAGE_NAME=$(echo '${{ inputs.image }}' | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Log in to Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+
+    - name: Generate tag suffix
```

Comment:
We need to allow the user the ability to add additional tags, for instance they might use semantic release and want to also tag the release version (but not create a cache tag for that one, but would also create a development image)

Document this possible usecase.